Dockerfile

```diff
@@ -19,4 +24,9 @@ EXPOSE 26656 26657 1317 9090
 
 ENV CELESTIA_HOME /opt
 
+RUN adduser -D -u 1000 celestia \
+    && chown -R celestia:celestia /opt
```

Comment:
[Question]
What are the permissions given to the `/opt` folder? Does the user `celestia` has write permissions to `/opt/...`?
Because it will write to it being the default home

Reply:
Yes, a user has `rwx` permission to the folders he owns. I tested starting the container locally and had no issues starting the binary.
I also tested it with a custom directory, and in this case you need to ensure that the folder has the correct permissions (either owned by user `1000` or correct `rwx` permission) as it has not access to all folders as `root` has.

This should be documented so that people running the container don't run into issues.

Reply:
Cool :+1: Guess we should create an issue to document this somewhere, approving

Reply:
https://github.com/celestiaorg/docs/issues/508

Reply:
hey @smuu 
good point to bring ;) 

let me add some suggestions:

```suggestion

# User data
ENV UID=10001
ENV GID=10001
ENV USR_NAME=celestia
ENV CELESTIA_HOME /opt/${USR_NAME}

# Create the group and the user, provide GID & UID
RUN groupadd -g ${GID} ${USR_NAME} \
    && adduser -u ${UID} \
       --gid ${GID} \
       --disabled-login \
       --disabled-password \
       --gecos "" \
       --home ${CELESTIA_HOME}
       
USER ${USR_NAME}    
```

why don't use UID/GID below 10000?
*UIDs below 10,000 are a security risk on several systems, because if someone does manage to escalate privileges outside the Docker container their Docker container UID may overlap with a more privileged system user's UID granting them additional permissions. For best security, always run your processes as a UID/GID above 10,000.*

Reply:
we don't actually check 

> ENV CELESTIA_HOME /opt/${USR_NAME}

atm, but imo we should keep this and update the code to do so. I forget where, but I think we have this in some issue somewhere

Reply:
in this case it's just for using the env var later when creating the user `home` dir

```
RUN groupadd -g ${GID} ${USR_NAME} \
	    && adduser -u ${UID} \
	       --gid ${GID} \
	       --disabled-login \
	       --disabled-password \
	       --gecos "" \
	       --home ${CELESTIA_HOME} # here
```

regarding the other issue, it was because we have to specify the `--home` when executing the CLI every time, by default it tries to use the user home dir (currently root)
```
"/root/.celestia-app"
```

we'll have to test it, tho I'd say that with this change, running the container as `celestia` user, the `--home` flag should go to the user home dir `(/opt/${USR_NAME})`

Reply:
Good suggestion @jrmanes. Is there a reason, that you're using `ENV` instead of `ARG` here?

Reply:
> we don't actually check
> 
> > ENV CELESTIA_HOME /opt/${USR_NAME}
> 
> atm, but imo we should keep this and update the code to do so. I forget where, but I think we have this in some issue somewhere

@evan-forbes You mean this will not work with the current code and we would need to change the code to support this? (With this I mean `/opt/{USR_NAME}`.

Reply:
> Good suggestion @jrmanes. Is there a reason, that you're using `ENV` instead of `ARG` here?

nothing special, we can use `ARG` for all of them  👌

Reply:
@smuu yeah, we just have to add a line above this to look for the env var first https://github.com/celestiaorg/celestia-app/blob/da2e54b0ca48c7e85a2626c9b10a985706ae39fe/app/app.go#L178-L184

Reply:
> @evan-forbes Please check this: [b41b2e4](https://github.com/celestiaorg/celestia-app/commit/b41b2e4dc43ab9014df0e5ee3f81bca83aee5778)

left a comment/question for evan.